streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -440,7 +430,10 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
       rcvr
     }
 
-    runDummySparkJob()
+    while ((System.currentTimeMillis() - createTime) < maxRegisteredWaitingTimeMs) {}
```

Comment:
You're right. But I think it only waste a little time,
and how to write code gracefully? 
I hope to make it better but do not know how to do it.